File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetwork.java
package com.hypnoweb.hypnowebapp.catalog.repo.network;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by stefan on 12/9/2017.
 */

public class CatalogNetwork {

    @SerializedName("records")
    private final ArrayList<MovieNetwork> movies;

    CatalogNetwork() {
        movies = null;
    }

    public ArrayList<MovieNetwork> getMovies() {
        return movies;
    }

    static class MovieNetwork {
        @SerializedName("id")
        private String movieId;
        @SerializedName("title_vo")
        private String shortTitle;
        @SerializedName("title_vf")
        private String biggerTitle;
        @SerializedName("category_id")
        private String movieTypeId;
        @SerializedName("banner_url")
        private String banner;

        public String getMovieId() {
            return movieId;
        }

        public String getShortTitle() {
            return shortTitle;
        }

        public String getBiggerTitle() {
            return biggerTitle;
        }

        public String getMovieTypeId() {
            return movieTypeId;
        }

        public String getBanner() {
            return banner;
        }
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetwork.kt
package com.hypnoweb.hypnowebapp.catalog.repo.network

import com.google.gson.annotations.SerializedName
import java.util.*


class CatalogNetwork internal constructor(@SerializedName("records") val movies: ArrayList<MovieNetwork>? = null) {

    data class MovieNetwork constructor(@SerializedName("id") val movieId: String? = null,
                                        @SerializedName("title_vo") val shortTitle: String? = null,
                                        @SerializedName("title_vf") val biggerTitle: String? = null,
                                        @SerializedName("category_id") val movieTypeId: String? = null,
                                        @SerializedName("banner_url") val banner: String? = null)
}
